server/test/unit/server/db/model/test_dispatch.py

```diff
@@ -614,6 +614,28 @@ def test_returns_dict(self):
 
         self.assertTrue(isinstance(call.as_dict(), dict))
 
+    def test_mongoengine_objects_serialized(self):
+        """
+        Ensure that mongoengine objects in the scheduled call were serialized.
+        """
+        schedule = bson.SON(SCHEDULE)
+        call = ScheduledCall.from_db(schedule)
+        call['principal'] = model.User('test_user')
+
+        result = call.as_dict()
+        self.assertTrue(isinstance(result['principal'], dict))
```

Comment:
Assuming `model.User` is the only mongoengine object, it's probably enough to just make sure `result['principal']` was serialized properly.